pages/dogs/index.js

```diff
@@ -1,3 +1,46 @@
+import useSWR from "swr";
+import { useRouter } from "next/router";
+import LoadingSpinner from "@/components/LoadingSpinner";
+import AllDogsCard from "@/components/AllDogsCard";
+import styled from "styled-components";
+import Headline from "@/components/Headline";
+
 export default function Dogs() {
-  return <div>dogs</div>;
+  // get route from URL
+  const router = useRouter();
+
+  const { data: dogs, isLoading } = useSWR(
+    router.isReady ? `/api/dogs?` : null
+  );
+
```

Comment:
```suggestion
  const { data: dogs, isLoading } = useSWR(`/api/dogs`);
```

In the context of the provided code, if the URL for fetching the dog data (/api/dogs) doesn't depend on any router parameters (like query strings or dynamic route segments), then waiting for router.isReady before making the API call is unnecessary.

The router.isReady condition is typically used in scenarios where the API endpoint or the data you're fetching depends on parameters from the URL. For example, if you're fetching details for a specific dog and the dog's ID comes from the URL (like /dogs/[id]), you'd need to wait for router.isReady to ensure that the ID parameter is available.

Also the questionmark can be removed from the api route string